splunk/provider.go

```diff
@@ -76,6 +76,7 @@ func providerResources() map[string]*schema.Resource {
 		"splunk_authentication_users":        authenticationUsers(),
 		"splunk_authorization_roles":         authorizationRoles(),
 		"splunk_global_http_event_collector": globalHttpEventCollector(),
+		"splunk_generic_acl":                 genericAcl(),
```

Comment:
I chose this name to be similar to how the Vault provider names some resources. This resource could be used for pretty much anything you want to apply an ACL on, but as such it requires a bit of thinking to do so properly.

My thought is I would also add a `splunk_app_acl`, `splunk_view_acl`, etc. for higher frequency use cases which would be more user friendly.

Reply:
Yeah, sounds good.